### ISSUE TITLE
412: Invalid attachment stub in [doc id] for [attachment name]

### ISSUE THREAD

**Issue by brendand** [couchbase/couchbase-lite-ios#1248]
Since the latest commit, I've been having this error that won't seem to go away.

```
0 : error = <CFString 0x618000a2cc40 [0x7fff7a404440]>{contents = "missing_stub"}
2 : reason = <CFString 0x6180004ab940 [0x7fff7a404440]>{contents = "Invalid attachment stub in frm-7ef258070e0e45f8a4d11958d5e151c2 for icon"}

```

The document in question I have already deleted. I also compacted the database. But it seems to just want to keep syncing this icon attachment. `icon` is the name of the attachment that I added to my document.

The request URL is:

https://tapzapp.cloudant.com/db-0ed52ecc24b24b688ca3cf7a9341022b/_bulk_docs

It's a POST request.

Perhaps it's a new issue related to commit 955e8c671507446707396f5e78a6ff9af52dd13d

--------
* Version:   1.3 beta commit # f18ecd3748ae0d731e6b54e48b27e143ffffa2a2
* Client OS: OS X 10.11.5
* Server:    IBM Cloudant

**Comment by snej** [comment]
Hm. Could you capture the request body of that POST?

**Comment by brendand** [comment]
I'm still digging into this. I just found something really weird where the `_id` property of my document is actually `frm-7ef258070e0e45f8a4d11958d5e151c2 for icon` which is totally bogus. Somehow that `for icon` was appended to the ID. This may end up being something else wrong. I'll update or close this if I find out the real cause of that. Very weird.

**Comment by brendand** [comment]
No. That was just a parsing error of mine. Sorry for the confusion. Ignore these last 2 comments. :)

**Comment by brendand** [comment]
Ok, once I fixed my parsing error I found that it was coming from a document that I had not actually deleted. However, once I set the icon on my form to something else, it synced ok. Now I'm just back to the other `400 bad request` error with the `attachment longer than expected` error. I've already submitted a bug report to Cloudant for that one, but have not yet heard from them.

I'm going to close this one for now. If I see it again I'll re-open and provide the request data.
